qt/src/SourceManager.cc

```diff
@@ -184,6 +185,20 @@ bool SourceManager::checkTextLayer(const QString& filename) const {
 	return true;
 }
 
+void SourceManager::extractAdditionalInfo(const QString& filename, Source* source) const {
+	if(filename.endsWith(".pdf", Qt::CaseInsensitive)) {
+		std::unique_ptr<Poppler::Document> document(Poppler::Document::load(filename));
```

Comment:
The document is now opened three times when loading the source, once for the password, once for checking text, and once for the info. This is not very efficient, the document should only be opened once.